portal-impl/test/integration/com/liferay/portlet/service/persistence/GroupFinderTest.java

```diff
@@ -170,6 +181,44 @@ public void testFindByCompanyId() throws Exception {
 	}
 
 	@Test
+	public void testFindByCompanyIdByUserGroupGroup() throws Exception {
+		_userGroup = UserGroupTestUtil.addUserGroup();
+		_userGroupGroup = GroupTestUtil.addGroup();
+		_userGroupUser = UserTestUtil.addUser();
+
+		GroupLocalServiceUtil.addUserGroupGroup(
```

Comment:
Hey @pei-jung,

Is this really necessary? When creating a UserGroup it automatically gets an associated Group. You should not have to manually create one and associate it.